ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go

```diff
@@ -2175,3 +2183,297 @@ func TestCheckNetworkPacketLoss(t *testing.T) {
 	tcs := generateCheckNetworkPacketLossTestCases()
 	testNetworkFaultInjectionCommon(t, tcs, NetworkFaultPath(types.PacketLossFaultType, types.CheckNetworkFaultPostfix))
 }
+
+func TestNetworkFaultRequestOrdering(t *testing.T) {
```

Comment:
nit (non-blocking): There seems to be a lot of repeated code across test cases, especially between L2242-2288 and L2289-L2335. This is not a strict blocker for me but I would prefer if we could look to minimize the amount of repeated code across test cases (for code maintainability purposes, particularly if we expect to add any new fault types in the future) and factor those out into for loop.